packages/smart-wallet/src/offerWatcher.js

```diff
@@ -274,7 +274,20 @@ export const prepareOfferWatcher = (baggage, vowTools) => {
       resultWatcher: {
         onFulfilled(result) {
           const { facets } = this;
-          facets.helper.publishResult(result);
+          const { walletHelper } = this.state;
+          const { after } = this.state.status;
+          if (after) {
+            if (after.saveAs) {
```

Comment:
surprising that `after: {}` will cause the result to be neither saved nor published.